connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TaskConfig.java

```diff
@@ -49,6 +49,6 @@ public TaskConfig() {
     }
 
     public TaskConfig(Map<String, ?> props) {
-        super(config, props);
+        super(config, props,true);
```

Comment:
Hmm, so I realized the issue with this is that you lose the context of which task (or connector if we fix the other instances) is even listing this config. With `ProducerConfig` or `ConsumerConfig`, just listing the classname in `logAll()` usually works fine since you're often just creating one instance. Here, we'll just get `TaskConfig` in the output.

Do we think it's enough to just have the surrounding context from previous log lines? It does look like we log which task is being created pretty close before this. But there is also a `ConnectorConfig` that is parsed before the `TaskConfig`, so I'm not sure how obvious it'll be that the two are tied together. (MDC could probably help a lot here.)